File: P04 - Inherirance/Python/AnekaBidangDatar.py
# File AnekaBidangDatar

class Persegi:
    def __init__(self) -> None:
        self._panj_sisi: int = 0
        self._luas: int = 0
        
    @property
    def panj_sisi(self)-> int:
        return self._panj_sisi
    
    @panj_sisi.setter
    def panj_sisi(self, n:int) -> None:
        self._panj_sisi = n
        
    @property
    def luas(self) -> int:
        return self._luas
    
    def hitung_luas(self) -> int:
        self._luas = self._panj_sisi * self._panj_sisi
        
class PersegiPanjang(Persegi):
    def __init__(self) -> None:
        self.__lebar: int = 0
        
    @property
    def lebar_sisi(self) -> int:
        return self.__lebar_sisi
    
    @lebar_sisi.setter
    def lebar_sisi(self, n:int) -> None:
        self.__lebar_sisi = n
        
    def hitung_luas(self) -> int:
        self._luas = self.panj_sisi * self.__lebar_sisi
        
def main():
    p = Persegi()
    p.panj_sisi = 5
    p.hitung_luas()
    print("Bujur Sangkar:")
    print("Panj Sisi: " + str(p.panj_sisi))
    print("Luas: " + str(p.luas))
    print()
    
    pp = PersegiPanjang()
    pp.panj_sisi = 8
    pp.lebar_sisi = 4
    pp.hitung_luas()
    print("Persegi Panjang:")
    print("Panj Sisi: " + str(pp.panj_sisi))
    print("Lebar Sisi: " + str(pp.lebar_sisi))
    print("Luas: " + str(pp.luas))
    
if __name__ == "__main__":
    main()

File: P04 - Inherirance/Java/AnekaBidangDatar.java
package Praktikum.P04.P04c_Latihan_Inheritance_Java_227008_Elmosius_Suli;
// File : AnekaBidangDatar.java
// Nama : Elmosius Suli
// NRP  : 2272008
// Kelas : B
// Ket :

class Persegi{
    protected int panjSisi;
    protected int luas;

    public Persegi(){

    }

    public void setPanjSisi(int n){
        this.panjSisi = n;
    }
    public int getPanjSisi(){
        return this.panjSisi;
    }
    public int getLuas(){
        return luas;
    }

    public void hitungLuas(){
        this.luas = (int)Math.pow(this.panjSisi, 2);
    }
}

class PersegiPanjang extends Persegi{
    private int lebarSisi;

    public PersegiPanjang(){

    }
    public void setLebarSisi(int n){
        this.lebarSisi = n;
    }
    public int getLebarSisi(){
        return this.lebarSisi;
    }
    public void hitungLuas(){
        this.luas = super.panjSisi * this.lebarSisi;
    }
}

public class AnekaBidangDatar{
    public static void main(String[] args) {
        Persegi p = new Persegi();
        p.setPanjSisi(5);
        p.hitungLuas();
        System.out.println("Bujur Sangkar:");
        System.out.println("Panj Sisi: " + p.getPanjSisi());
        System.out.println("Luas: " + p.getLuas());
        System.out.println();

        PersegiPanjang pp = new PersegiPanjang();
        pp.setPanjSisi(8);
        pp.setLebarSisi(4);
        pp.hitungLuas();
        System.out.println("Persegi Panjang:");
        System.out.println("Panj Sisi: " + pp.getPanjSisi());
        System.out.println("Lebar Sisi: " + pp.getLebarSisi());
        System.out.println("Luas: " + pp.getLuas());
        System.out.println();
    }

}
